src/admin/components/forms/field-types/Relationship/index.tsx

```diff
@@ -277,79 +209,64 @@ const Relationship: React.FC<Props> = (props) => {
-            where: {
-              id: {
-                in: ids,
-              },
+      const idsToLoad = ids.filter((id) => {
+        return !options.find((optionGroup) => optionGroup?.options?.find((option) => option.value === id && option.relationTo === relation));
+      });
+
+      if (idsToLoad.length > 0) {
+        const query = {
+          where: {
+            id: {
+              in: idsToLoad,
             },
-            depth: 0,
-            limit: ids.length,
-          };
+          },
+          depth: 0,
+          limit: idsToLoad.length,
+        };
 
-          if (!errorLoading) {
-            const response = await fetch(`${serverURL}${api}/${relation}?${qs.stringify(query)}`, {
-              credentials: 'include',
-              headers: {
-                'Accept-Language': i18n.language,
-              },
-            });
-            const collection = collections.find((coll) => coll.slug === relation);
-            if (response.ok) {
-              const data = await response.json();
-              dispatchOptions({ type: 'ADD', docs: data.docs, hasMultipleRelations, collection, sort: true, ids, i18n });
-            } else if (response.status === 403) {
-              dispatchOptions({ type: 'ADD', docs: [], hasMultipleRelations, collection, sort: true, ids, i18n });
-            }
+        if (!errorLoading) {
+          const response = await fetch(`${serverURL}${api}/${relation}?${qs.stringify(query)}`, {
+            credentials: 'include',
+            headers: {
+              'Accept-Language': i18n.language,
+            },
+          });
+          const collection = collections.find((coll) => coll.slug === relation);
+          if (response.ok) {
+            const data = await response.json();
+            dispatchOptions({ type: 'ADD', docs: data.docs, collection, sort: true, ids, i18n });
+          } else if (response.status === 403) {
+            dispatchOptions({ type: 'ADD', docs: [], collection, sort: true, ids, i18n });
           }
         }
-      }, Promise.resolve());
-
-      setHasLoadedValueOptions(true);
-    }
-  }, [hasMany, hasMultipleRelations, relationTo, initialValue, hasLoadedValueOptions, errorLoading, collections, api, serverURL, i18n]);
-
-  useEffect(() => {
-    if (!filterOptions) return;
-
-    const [data, siblingData] = getFormData();
-
-    const newOptionFilters = getFilterOptionsQuery(filterOptions, {
-      id,
-      data,
-      relationTo,
-      siblingData,
-      user,
-    });
-    if (!equal(newOptionFilters, optionFilters)) {
-      setOptionFilters(newOptionFilters);
-    }
-  }, [relationTo, filterOptions, optionFilters, id, getFormData, path, user]);
-
-  useEffect(() => {
-    if (optionFilters || !filterOptions) {
-      setHasLoadedValueOptions(false);
-      getResults({
-        value: initialValue,
-      });
-    }
-  }, [initialValue, getResults, optionFilters, filterOptions]);
+      }
+    }, Promise.resolve());
+  }, [
+    options,
```

Comment:
`options` as a dependency here is throwing this effect into an infinite loop because it's being reset on lines 251 and 253.

Reply:
I was able to remove that dependency from the effect by moving the logic into the `optionsReducer`, where `state` replaces `options`. See this commit: https://github.com/payloadcms/payload/commit/54ef40a335905f7295f847c68762f7fe06bccc30